spec/services/event_log/move_runner_spec.rb

```diff
@@ -81,6 +80,36 @@
     end
   end
 
+  context 'when event_name=cancel' do
+    let!(:event) { create(:move_event, :cancel, eventable: move) }
+
+    context 'when the move is requested' do
+      it 'updates the move status to cancelled' do
+        expect { runner.call }.to change(move, :status).from('requested').to('cancelled')
+      end
+
+      it 'updates the move cancellation_reason' do
+        expect { runner.call }.to change(move, :cancellation_reason).from(nil).to('supplier_declined_to_move')
+      end
+
+      it 'updates the move cancellation_reason_comment' do
+        expect { runner.call }.to change(move, :cancellation_reason_comment).from(nil).to('computer says no')
+      end
+
+      it_behaves_like 'it calls the Notifier with an update_status action_name'
+    end
+
+    context 'when the move is already cancelled' do
+      let!(:move) { create(:move, :cancelled, cancellation_reason: 'supplier_declined_to_move', cancellation_reason_comment: 'computer says no') }
+
+      it_behaves_like 'it does not call the Notifier'
```

Comment:
👍

Reply:
can you add something similar when the move is invalid to check that we send no notifications?

Reply:
not sure I follow? lets chat when you have a moment